### PR TITLE
(117) Refactor technical naming for plans to journeys and questions to steps

### PR DESCRIPTION
**Pull request by tahb**
## Changes in this PR

- Refactor routing and modelling of "Plan" to "Journey" to abstract the concept in order to support plans *and* specifications
- Refactor routing and modelling of "Question" to "Step" to abstract the concept in order to support questions, static content *and* decisions.
- Fixtures for the contentful type `question` remains and so the fixtures have kept their names 
- `Journey#new` still connects to the concept of a "plan" through an environment variable. I imagine that when we want to add a different type of journey, the button a user clicks can send across some context for which journey they are attempting to start. The controller can then use that to use different `next_entry_id`.
- I assume that a `Step` can eventually be a Contentful type of `static_content`, which means our `Step` model would handle questions and content which require no associated `Answer` record.

### CODE REVIEW

**Comment by tahb** [comment]
mmm this release failed.. no logs in Heroku https://dashboard.heroku.com/apps/dfe-bfys-pr-56/activity/releases/855af7fd-2257-4a78-869c-4994819c21d8 but Rollbar points at a migration problem https://rollbar.com/dxw/dfe-buy-for-your-school/items/35/?utm_campaign=new_item_message&utm_medium=slack&utm_source=rollbar-notification!

**Comment by tahb** [comment]
A migration includes reference to a `Question` model that no longer exists. I'll have to patch those here.

**Comment by jacksonj04** [comment]
Patch for broken data migrations is in #57.

**Comment by tahb** [comment]
@jacksonj04 this is now rebasing ontop of that migration patch

File: spec/services/create_journey_step_spec.rb [hunk 1]
```diff
@@ -118,30 +118,30 @@
...
+            allowed_step_types: CreateJourneyStep::ALLOWED_CONTENTFUL_QUESTION_TYPES.join(", "))
           .and_call_original
         expect { described_class.new(journey: journey, contentful_entry: fake_entry).call }
-          .to raise_error(CreateJourneyQuestion::UnexpectedContentfulModel)
+          .to raise_error(CreateJourneyStep::UnexpectedContentfulModel)
       end
     end
 
-    context "when the new question has an unexpected type" do
+    context "when the new step has an question unexpected type" do
```
Comment (jacksonj04) [inline spec/services/create_journey_step_spec.rb:130]:
```suggestion
    context "when the new step has an unexpected step type" do
```

**Review by jacksonj04 — APPROVED** [review]
One tiny errant comment, otherwise this heckin chonk of a PR is good to go!

**Pull request merged**